pkg/connector/groups.go

```diff
@@ -259,6 +252,219 @@ func (g *groupBuilder) Grants(ctx context.Context, resource *v2.Resource, pToken
 	return rv, "", nil, nil
 }
 
+func (g *groupBuilder) Grant(ctx context.Context, principal *v2.Resource, entitlement *v2.Entitlement) (annotations.Annotations, error) {
+	l := ctxzap.Extract(ctx)
+
+	if !isValidPrincipal(principal.Id) {
+		l.Warn(
+			"databricks-connector: only users, groups and service principals can be granted group permissions",
+			zap.String("principal_id", principal.Id.String()),
+			zap.String("principal_type", principal.Id.ResourceType),
+		)
+
+		return nil, fmt.Errorf("databricks-connector: only users, groups and service principals can be granted group permissions")
+	}
+
+	groupTrait, err := rs.GetGroupTrait(entitlement.Resource)
+	if err != nil {
+		return nil, fmt.Errorf("databricks-connector: failed to get group trait: %w", err)
+	}
+
+	parentType, parentID, err := getParentInfoFromProfile(groupTrait.Profile)
+	if err != nil {
+		return nil, fmt.Errorf("databricks-connector: failed to get parent info from group profile: %w", err)
+	}
+
+	if parentType == workspaceResourceType.Id {
+		g.client.SetWorkspaceConfig(parentID)
+	} else {
+		g.client.SetAccountConfig()
+	}
+
+	groupID := entitlement.Resource.Id.Resource
+
+	if entitlement.Slug == groupMemberEntitlement {
```

Comment:
You could un-indent a lot of code by early returning here:

```suggestion
	if entitlement.Slug != groupMemberEntitlement {
		return nil, nil
	}
```

Or maybe it should error if the entitlement is something other than group membership? I'm not sure.

Reply:
The `else` branch is important in this case, but can be un-indented. I added comments to clarify more in code.